api/v1beta1/types_class.go

```diff
@@ -160,3 +160,8 @@ func (sgc *SecurityGroupClass) setDefaults() {
 		}
 	}
 }
+
+// setDefaults sets default values for NatGatewayClass.
+func (natgaw *NatGatewayClassSpec) setDefaults() {
+	natgaw.Name = DefaultNATGateway
```

Comment:
what is the "scope" of a NAT gateway name in Azure? I'm wondering if it makes sense to default the name to a hardcoded value for all clusters in a clusterclass. Notice how the other resources (Subnet, VNet, etc) are not getting a default name value since they need to have unique names withing a resource group, in some cases even within a subscription.

Reply:
Sorry, I didn't realize the NAT gateway should be unique. Thanks for your kind reminder!
>what is the "scope" of a NAT gateway name in Azure?

This NAT gateway name is for `AzureClusterTemplate` object.
Now, I have removed the `DefaultNATGateway` variable and generate a unique name by default.